compiler/rustc_typeck/src/collect.rs

```diff
@@ -2294,8 +2294,14 @@ fn gather_explicit_predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericP
                 }))
             }
 
-            hir::WherePredicate::EqPredicate(..) => {
-                // FIXME(#20041)
+            hir::WherePredicate::EqPredicate(where_eq_pred) => {
+                let hir::WhereEqPredicate { hir_id: _, span, lhs_ty, rhs_ty } = where_eq_pred;
+                let lhs = icx.to_ty(&lhs_ty);
+                let rhs = icx.to_ty(&rhs_ty);
+                // this is a cheap approximation of equality, by checking if both are subtypes
```

Comment:
TODO delete old comment of me being lazy.
Here we can also do something like check that both types aren't params, which doesn't do much for now.